packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -674,22 +674,32 @@ export class RunningSummarizer implements IDisposable {
 						});
 						await delay(delaySeconds * 1000);
 					}
-				}
 
-				if (this.shouldAbortOnSummaryFailure) {
-					this.mc.logger.sendTelemetryEvent(
-						{
-							eventName: "ClosingSummarizerOnSummaryStale",
-							reason,
-							message: lastResult?.message,
-						},
-						lastResult?.error,
-					);
+					if (this.shouldAbortOnSummaryFailure) {
+						// There's no need to delay here as we already have summarizeAttemptDelay given to us by the server.
+						if (delaySeconds === 0) {
+							const closeSummarizerDelayOverrideMs =
+								this.mc.config.getNumber(
+									"Fluid.ContainerRuntime.Test.CloseSummarizerOnSummaryStaleDelayOverrideMs",
+								) ?? defaultCloseSummarizerDelayMs;
+							await delay(closeSummarizerDelayOverrideMs);
```

Comment:
Any reason not to set `delaySeconds` above let code above do the wait?
It feels like there is a lot of special casing / duplication of existing code that we can avoid by plugging into existing workflow that already does all we need - delays and closing container.

Reply:
The more I'm advised to make modifications, the more I want to gut this code here and just keep it simple. I don't really want to maintain the same code in two different places. I'd rather go back to the implementation, where all the logic was in one class.